model/lifecycle.go

```diff
@@ -877,6 +880,39 @@ func createTasksForBuild(creationInfo TaskCreationInfo) (task.Tasks, error) {
 	return tasks, nil
 }
 
+// checkUsersPatchTaskLimit takes in an input list of tasks that is set to get activated, and checks if they're
+// patch tasks, and that the request has been submitted by a user. If so, the maximum hourly patch tasks counter
+// will be incremented accordingly. the addExecutionTasks parameter indicates that execution tasks are included
+// as part of the input tasks param, otherwise we need to account for them.
+func checkUsersPatchTaskLimit(requester, username string, addExecutionTasks bool, tasks ...task.Task) error {
+	// we only care about patch tasks that are to be activated by an actual person
+	if !evergreen.IsPatchRequester(requester) || evergreen.IsSystemActivator(username) {
+		return nil
+	}
+	numTasksToActivate := 0
+	for _, t := range tasks {
+		if t.Activated {
+			if addExecutionTasks && t.DisplayOnly {
+				execTaskIdsToRestart, err := findExecTasksToReset(&t)
+				if err != nil {
+					return errors.Wrap(err, "finding execution tasks to restart")
+				}
+				numTasksToActivate += len(execTaskIdsToRestart)
+			}
+			numTasksToActivate++
+		}
+	}
+	u, err := user.FindOneById(username)
+	if err != nil {
+		return errors.Wrap(err, "getting user")
+	}
+	if u != nil {
+		s := evergreen.GetEnvironment().Settings()
+		return errors.Wrapf(u.CheckAndUpdateSchedulingLimit(s, numTasksToActivate), "checking task scheduling limit for user '%s'", u.Id)
```

Comment:
It looks like there are places we call this directly -- it seems like we should always be checking that "ifPatchRequester" logic. Did we audit these?